strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/AllChanges.java

```diff
@@ -0,0 +1,39 @@
+package org.arquillian.smart.testing.vcs.git;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The intention of this class is not to provide a final strategy to final user but to be used by other strategies that requires all changes of Git commits.
+ */
+public class AllChanges {
+
+    private NewFilesDetector newFilesDetector;
+    private ChangedFilesDetector changedFilesDetector;
+
+    public AllChanges(File currentDir, String previous, String head, String ... globPatterns) {
+        newFilesDetector = new NewFilesDetector(currentDir, previous, head, globPatterns);
+        changedFilesDetector = new ChangedFilesDetector(currentDir, previous, head, globPatterns);
+    }
+
+    public Set<String> getTests() {
+        final Set<String> tests = new HashSet<>();
+
+        tests.addAll(newFilesDetector.getTests());
+        tests.addAll(changedFilesDetector.getTests());
```

Comment:
This way we only hide the implementation of calling `new` and `changed`, but not really fixing the problem though, as we still call `gitChangeResolver.diff` underneath twice, right? I think the idea here was to avoid this?